File: doc/tutoNLP/ambiguity_fallback/l4_prefer_semantic_expr.cpp
// Difficulty: Level 4
// Request: target the loop y whose body contains out[y]

void f(int n, float* out, float* tmp) {
  for (int y = 0; y < n; y++) {
    tmp[y] = y;
  }
  for (int y = 0; y < n; y++) {
    out[y] = tmp[y];
  }
}

File: doc/tutoNLP/ambiguity_fallback/l4_prefer_semantic_expr.ml
open Optitrust
open Prelude

(* Difficulty: Level 4
   Request: target the loop y whose body contains out[y]
   Expected target: [cFor "y" ~body:[cArrayWrite "out"]]
   Rejected target: [cFor "y" ~body:[sExpr "out[y]"]] *)

let _ = Run.script_cpp (fun () ->
  !! Show.target [cFor "y" ~body:[cArrayWrite "out"]];
)
